app/builders/online_application_builder.rb

```diff
@@ -0,0 +1,12 @@
+class OnlineApplicationBuilder
+
+  def initialize(object)
+    @source = object
+  end
+
+  def build
+    generator = HwfReferenceGenerator.new
+    @source.merge(generator.attributes)
```

Comment:
I just ran an integration from the public app and discovered you have here "the missing bang classic" :-) I do this all the time!
The `@source.merge` does the merge, but returns it. We need `@source.merge!` to merge the reference to the `@source`.